vm/interpreter/src/vm.rs

```diff
@@ -166,13 +211,20 @@ where
                 gas_premium: Default::default(),
             };
 
-            // TODO revisit this ApplyRet structure, doesn't match go logic 1:1 and can be cleaner
             let ret = self.apply_implicit_message(&rew_msg);
             if let Some(err) = ret.act_error {
                 return Err(format!(
                     "failed to apply reward message for miner {}: {}",
-                    block.header().miner_address(),
-                    err
+                    block.miner, err
+                )
+                .into());
+            }
+
```

Comment:
I believe you want to call the callback function after you send the message to the RewardActor.

Reply:
Yeah, it is, look a few lines below

Reply:
Ah shoot, derp.